dexmaker-tests/src/androidTest/java/com/android/dx/stock/ProxyBuilderTest.java

```diff
@@ -1239,4 +1243,34 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
         assertEquals("fake E", proxy.returnE());
 
     }
+
+    @Test
+    public void testImplementingDifferentInterfacesWithSharedClassLoader() throws IOException {
+        assumeTrue(Build.VERSION.SDK_INT >= 24);
+
+        Class<?> c1 = ProxyBuilder.forClass(SimpleClass.class)
+                .implementing(Runnable.class).withSharedClassLoader().buildProxyClass();
+        Class<?> c2 = ProxyBuilder.forClass(SimpleClass.class)
+                .implementing(Callable.class).withSharedClassLoader().buildProxyClass();
+        assertNotSame(c1, c2);
+    }
+
+    @Test
+    public void testImplementingDifferentInterfacesWithDexMakerCaching() throws Exception {
+        proxyFor(SimpleClass.class).implementing(Runnable.class).buildProxyClass();
+        assertEquals(1, getGeneratedProxyClasses().size());
+        assertEquals(2, versionedDxDir.listFiles().length);
+
+        proxyFor(SimpleClass.class).implementing(Callable.class).buildProxyClass();
+        assertEquals(2, getGeneratedProxyClasses().size());
+        assertEquals(3, versionedDxDir.listFiles().length);
+
+        Map<?, ?> cacheCopy = new HashMap<>(getGeneratedProxyClasses());
+        Set<?> generatedFiles = new HashSet<>(Arrays.asList(versionedDxDir.listFiles()));
+
+        proxyFor(SimpleClass.class).implementing(Runnable.class).buildProxyClass();
+        proxyFor(SimpleClass.class).implementing(Callable.class).buildProxyClass();
+        assertEquals(cacheCopy, getGeneratedProxyClasses());
```

Comment:
Should't we check that the proxy classes are not the same?

Reply:
Deleted a test to check the cache